tests/functional/z_cluster/test_enable_multus_existing_cluster.py

```diff
@@ -0,0 +1,244 @@
+"""
+Test module for enabling Multus networking on an existing OCS cluster.
+"""
+
+import logging
+import json
+import tempfile
+
+from ocs_ci.framework import config
+from ocs_ci.framework.testlib import polarion_id
+
+from ocs_ci.utility.reporting import get_polarion_id
+from ocs_ci.framework.testlib import (
+    ManageTest,
+    tier2,
+    skipif_managed_service,
+    skipif_external_mode,
+    brown_squad,
+)
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.resources.storage_cluster import (
+    get_storage_cluster,
+    verify_multus_network,
+    verify_storage_cluster,
+)
+from ocs_ci.ocs.resources.pod import get_osd_pods, wait_for_storage_pods
+from ocs_ci.utility import templating
+from ocs_ci.utility.utils import run_cmd, TimeoutSampler
+
+
+logger = logging.getLogger(__name__)
+
+
+@brown_squad
+@skipif_managed_service
+@skipif_external_mode
+@polarion_id(get_polarion_id())
+class TestEnableMultusExistingCluster(ManageTest):
+    """
+    Test class for enabling Multus on an existing cluster.
+    """
+
+    @tier2
+    def test_enable_multus_on_existing_cluster(self):
+        """
+        Test enabling Multus networking on an existing OCS cluster.
+
+        Steps:
+        1. Verify StorageCluster is ready
+        2. Verify OSD pods are up and running
+        3. Create NetworkAttachmentDefinition using existing templates
+        4. Update StorageCluster to use Multus networking
+        5. Wait for OSD pods to restart with Multus configuration
+        6. Verify Multus networking is configured on pods
+        """
+        logger.info("Starting test to enable Multus on existing cluster")
+
+        logger.info("Step 1: Verifying StorageCluster is ready")
+        verify_storage_cluster()
+        logger.info("StorageCluster is Ready")
+
+        logger.info("Step 2: Verifying OSD pods are running")
+        wait_for_storage_pods()
+        osd_pods = get_osd_pods()
+        logger.info(f"All {len(osd_pods)} OSD pods are running")
+
+        sc = get_storage_cluster()
+
+        logger.info("Step 3: Creating Multus NetworkAttachmentDefinitions")
+
+        create_public_net = config.ENV_DATA.get("multus_create_public_net", True)
+        create_cluster_net = config.ENV_DATA.get("multus_create_cluster_net", True)
+
+        # Create Public Network
+        if create_public_net:
+            logger.info("Creating Multus public network")
+            nad_to_load = constants.MULTUS_PUBLIC_NET_YAML
+            if config.DEPLOYMENT.get("ipv6"):
+                nad_to_load = constants.MULTUS_PUBLIC_NET_IPV6_YAML
+
+            public_net_data = templating.load_yaml(nad_to_load)
+            public_net_data["metadata"]["name"] = config.ENV_DATA.get(
+                "multus_public_net_name"
+            )
+            public_net_data["metadata"]["namespace"] = config.ENV_DATA.get(
+                "multus_public_net_namespace"
+            )
+
+            public_net_config_str = public_net_data["spec"]["config"]
```

Comment:
Since this feature is specific to vSphere and Bare Metal, we prefer not to add configuration files via `--ocsci-conf` to the `run-ci` command. Also, because we currently have only one Bare Metal setup, I think we should run it only on vSphere. We can add the `@vsphere_platform_required` marker to the test.

CC @petr-balogh @shyRozen